### PR TITLE
Draft: Add insights-operator-gathering-conditions-service reference

### PR DESCRIPTION
**Pull request by juandspy**
To be merged after [CCXDEV-6760](https://issues.redhat.com/browse/CCXDEV-6760) completion.

### CODE REVIEW

**Review by karelhala — APPROVED** [review]

**Comment by juandspy** [comment]
@karelhala Any idea about what's going on with my api names? TY in advance
```
Error in ccx app definition.
 [Error [ValidationError]: "api.subItems.aggregator.apiName" is not allowed] {
  _original: { title: 'OpenShift Insights', api: { subItems: [Object] } },
  details: [
    {
      message: '"api.subItems.aggregator.apiName" is not allowed',
      path: [Array],
      type: 'object.unknown',
      context: [Object]
    }
  ]
}
```

**Comment by Hyperkid123** [comment]
> @karelhala Any idea about what's going on with my api names? TY in advance

@juandspy this format is not supported in the main.yml validator. @karelhala is this object structure supported?

**Comment by juandspy** [comment]
> @juandspy this format is not supported in the main.yml validator. @karelhala is this object structure supported?

@Hyperkid123 it's probably related with the `apiName` field as the rest is similar to [this one](https://github.com/RedHatInsights/cloud-services-config/blob/509b89bd1d976d4365f4d80ee7b1f8a0c5f0f533/main.yml#L404) that is fine.

**Comment by Hyperkid123** [comment]
@juandspy yes the issue is with the `apiName`. It is not included in the schema validator, only the `title` and `versions` are. The API schema is used to generate UI and I don't think the `apiName` attribute in the sub-items is used for anything.

**Comment by karelhala** [comment]
Aaaah, yeah! The `apiName` is not supported. The name is the key in the array of objects.

File: main.yml [hunk 1]
```diff
@@ -254,9 +254,17 @@ catalog:
 ccx:
   title: OpenShift Insights
   api:
-    versions:
-      - v1
-    apiName: insights-results-aggregator
+    subItems:
+      aggregator:
```
Comment (karelhala) [inline main.yml]:
```suggestion
      insights-results-aggregator:
```

File: main.yml [hunk 2]
```diff
@@ -254,9 +254,17 @@ catalog:
...
+        title: Insights Results Aggregator
+        versions:
+          - v1
+        apiName: insights-results-aggregator
```
Comment (karelhala) [inline main.yml]:
```suggestion
```

File: main.yml [hunk 3]
```diff
@@ -254,9 +254,17 @@ catalog:
...
+      gatheringconditions:
```
Comment (karelhala) [inline main.yml]:
```suggestion
      insights-operator-gathering-conditions-service:
```

File: main.yml [hunk 4]
```diff
@@ -254,9 +254,17 @@ catalog:
...
+        title: Insights Operator Gathering Conditions Service
+        versions:
+          - v1
+        apiName: insights-operator-gathering-conditions-service
```
Comment (karelhala) [inline main.yml]:
```suggestion
```

**Review by karelhala — COMMENTED** [review]
The apiName is not supported for nested api object. It was previously introduced to rename certain APIs (for instance if user had bookmarked it previously and the app name changed)

**Comment by juandspy** [comment]
@karelhala thanks a lot!

**Review by karelhala — APPROVED** [review]

**Comment by juandspy** [comment]
Hi @karelhala @Hyperkid123 . I've checked the stage documentation and it doesn't seem to wor. Some questions:
1. I don't understand how API documentation works. These links both work for the prod url:
    - https://console.redhat.com/docs/a_veryyyyyyyyyyyyyyyyyyy_random_string/insights
    - https://console.redhat.com/docs/api/insights  (this is the default)
2. If I try to access any API in the stage environment it says `No API definition provided`.

Any idea what's going on?

**Comment by Hyperkid123** [comment]
@juandspy It seems to be working

screenshot-console stage redhat com-2022 01 18-13_08_42

**Comment by juandspy** [comment]
Yes @Hyperkid123 . Can you please enter one of those APIs? It says `No API definition provided`.

**Comment by karelhala** [comment]
@juandspy The thing with the random string is that we have a redirect deep within Akamai, we should however fix it as it's a most certainly a bug.

The https://console.redhat.com/docs/api/insights is for advisor/insights, what you want to access is the ccx one https://console.stage.redhat.com/docs/api/insights-results-aggregator I am able to access this one. If I click on the `Insights Operator Gathering Conditions Service` it shows `No API definition provided.` because requesting URL https://console.stage.redhat.com/api/insights-operator-gathering-conditions-service/v1/openapi.json yields 404. Is the backend app deployed and properly setup? Moreover, is the URL correct?

**Comment by Hyperkid123** [comment]
> Yes @Hyperkid123 . Can you please enter one of those APIs? It says `No API definition provided`.

Seems to be working as well

screenshot-console stage redhat com-2022 01 18-13_12_34

**Comment by Hyperkid123** [comment]
Yeah as @karelhala https://console.stage.redhat.com/api/insights-operator-gathering-conditions-service/v1/openapi.json returns 404. So the UI has nothing to render.

**Comment by juandspy** [comment]
@Hyperkid123 Does the rest of the APIs return something different than 404? For the gathering conditions service it is expected. But I have the same behavior for the rest of the APIs.

**Comment by Hyperkid123** [comment]
Yeah on the details page we actually need the `openapi.json` file in order to render the swagger UI. It won't work without it.

**Comment by juandspy** [comment]
Alright. I wasn't able to see it. Everything fine in that case. TYVM!

**Comment by juandspy** [comment]
Hey @Hyperkid123 @karelhala . It seems that the API docs are not working yet: 
image
I would expected a dropdown menu such as the one for the "Drift" option. Any idea what's going on? Am I checking the right URL?

**Comment by Hyperkid123** [comment]
@juandspy the fixed sync job ate your changes. Let me put it back. You want this all the way to prod stable? Or stage is fine?

**Comment by juandspy** [comment]
> @juandspy the fixed sync job ate your changes. Let me put it back. You want this all the way to prod stable? Or stage is fine?

thanks @Hyperkid123 . Could you please notify me when the changes are on stage?

**Comment by juandspy** [comment]
> @Hyperkid123 You want this all the way to prod stable? 

Yes, please :+1:

**Comment by Hyperkid123** [comment]
@juandspy the changes are visible in stage beta: https://console.stage.redhat.com/beta/docs/api

Let me know if its OK and we will deploy it to prod-stable and the rest of the envs

**Comment by juandspy** [comment]
> @juandspy the changes are visible in stage beta: https://console.stage.redhat.com/beta/docs/api
> 
> Let me know if its OK and we will deploy it to prod-stable and the rest of the envs

Thank you @Hyperkid123. I guess it's ok, although all the services have a "No API definition provided.". Is this expected?

**Comment by Hyperkid123** [comment]
@juandspy this one works: https://console.stage.redhat.com/beta/docs/api/insights-results-aggregator 

The other one does not: https://console.stage.redhat.com/beta/docs/api/insights-operator-gathering-conditions-service and that is because this API call returns a 404: https://console.stage.redhat.com/api/insights-operator-gathering-conditions-service/v1/openapi.json

If we the OpenAPI spec does not exists, then it will not work

**Comment by juandspy** [comment]
@Hyperkid123 I'm having this error in any service:
image

@Hyperkid123 BTW, the gathering conditions endpoint is failing because the OpenAPI spec is under:
- https://console.stage.redhat.com/api/gathering/openapi.json

instead of 

- https://console.stage.redhat.com/api/insights-operator-gathering-conditions-service/v1/openapi.json

Is it possible to override it somehow? It would be a bit dangerous to change our API as we would need some effort from customers.

**Comment by karelhala** [comment]
@juandspy yes, you can change this line https://github.com/RedHatInsights/cloud-services-config/blob/ci-beta/main.yml#L222 to `gathering` and that'll be it.

**Comment by juandspy** [comment]
> @juandspy yes, you can change this line https://github.com/RedHatInsights/cloud-services-config/blob/ci-beta/main.yml#L222 to `gathering` and that'll be it.

@karelhala although we don't support versioning of the API?

**Comment by karelhala** [comment]
@juandspy will the API ever support versioning? We are supporting just APIs that are versioned. If you'd want to support API without versioning we'd have to open a JIRA ticket and work that in (probably after summit).

**Comment by juandspy** [comment]
Thanks @karelhala . We will update the API so that it supports versioning. Will come back to you once it's done so that we can check stage then move to prod.

**Comment by juandspy** [comment]
FYI, the correction with the versioned endpoints is https://github.com/RedHatInsights/cloud-services-config/pull/1152

**Pull request merged**